conan/tools/microsoft/visual.py

```diff
@@ -166,9 +169,6 @@ def _vcvars_vers(conanfile, compiler, vs_version):
         assert compiler == "msvc"
         # Code similar to CMakeToolchain toolset one
         compiler_version = str(conanfile.settings.compiler.version)
-        version_components = compiler_version.split(".")
-        assert len(version_components) >= 2  # there is a 19.XX
-        minor = version_components[1]
-        # The equivalent of compiler 19.26 is toolset 14.26
-        vcvars_ver = "14.{}".format(minor[0])
+        # The equivalent of compiler 192 is toolset 14.2
+        vcvars_ver = "14.{}".format(compiler_version[-1])
```

Comment:
Same comment as above. Need to take care of special cases for VS 16.9 and VS 16.11.